### PR TITLE
Fix: Remove unnecessary async state machine generation

### LINKED ISSUE #9871: Avoid unnecessary async state machine generation
We have plenty of code like below:

```csharp
async IAsyncOperation<T> Foo()
{
    return await AsyncInfo.Run(async () => 
    {
        // no await in the body at all
        return foo;
    });
}
```

which doesn't actually await anything in an async method. Instead, we should rewrite them to

```csharp
IAsyncOperation<T> Foo()
{
    return Task.FromResult(foo).AsAsyncOperation();
}
```

This can avoid unnecessary async state machine be generated to reduce GC pressure.

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9871 

**Validation**
How did you test these changes?
- [*] Built and ran the app

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs [hunk 1]
```diff
@@ -143,18 +140,15 @@ public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(Thumbnai
...
-					var name = IO.Path.GetFileName(path);
-					return new NativeStorageFile(path, name.Substring(name.LastIndexOf(":") + 1), DateTime.Now);
-				}
-				return null;
-			});
-		}
+				var name = IO.Path.GetFileName(path);
+				return Task.FromResult((BaseStorageFile)new NativeStorageFile(path, name[(name.LastIndexOf(":") + 1)..], DateTime.Now)).AsAsyncOperation();
+			}
+            return null;
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs:150]: They are not equivalent, instead using

```suggestion
            return Task.FromResult<BaseStorageFile>(null).AsAsyncOperation();
```

File: src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs [hunk 2]
```diff
@@ -112,23 +112,20 @@ public static async Task<bool> CheckDefaultZipApp(string filePath)
...
-					if (CheckAccess(containerPath))
-					{
-						return new ZipStorageFolder(path, containerPath);
-					}
+					return Task.FromResult((BaseStorageFolder)new ZipStorageFolder(path, containerPath)).AsAsyncOperation();
 				}
-				return null;
-			});
+			}
+			return null;
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs:128]: They are not equivalent, instead using

```suggestion
			return Task.FromResult<BaseStorageFolder>(null).AsAsyncOperation();
```

File: src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs [hunk 3]
```diff
@@ -112,23 +112,20 @@ public static async Task<bool> CheckDefaultZipApp(string filePath)
...
-			return AsyncInfo.Run<BaseStorageFolder>(async (cancellationToken) =>
+			if (!FileExtensionHelpers.IsBrowsableZipFile(path, out var ext))
 			{
-				if (!FileExtensionHelpers.IsBrowsableZipFile(path, out var ext))
-				{
-					return null;
-				}
-				var marker = path.IndexOf(ext, StringComparison.OrdinalIgnoreCase);
-				if (marker is not -1)
+				return null;
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs:117]: They are not equivalent, instead using

```suggestion
				return Task.FromResult<BaseStorageFolder>(null).AsAsyncOperation();
```

**Review by hez2010 — CHANGES_REQUESTED** [review]
Otherwise LGTM.

**Comment by hecksmosis** [comment]
That should be it then.

File: src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs [hunk 4]
```diff
@@ -143,18 +140,15 @@ public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(Thumbnai
 		public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(ThumbnailMode mode, uint requestedSize, ThumbnailOptions options)
 			=> throw new NotSupportedException();
 
-		public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
+		public static IAsyncOperation<BaseStorageFile>? FromPathAsync(string path)
```
Comment (jiejasonliu) [inline src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs:143]: with those changes, the nullable return doesn't make sense anymore right?

**Review by jentur-zabbeJ-8basdy — CHANGES_REQUESTED** [review]

**Comment by hecksmosis** [comment]
@hez2010 Is everything ok now?

File: src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs [hunk 5]
```diff
@@ -145,16 +142,13 @@ public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(Thumbnai
 
 		public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs:143]: Nullable: `IAsyncOperation<BaseStorageFile?>`

File: src/Files.App/Filesystem/StorageItems/ZipStorageFile.cs [hunk 6]
```diff
@@ -64,27 +64,24 @@ public override IAsyncOperation<StorageFile> ToStorageFileAsync()
 
 		public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/ZipStorageFile.cs:65]: Nullable: `IAsyncOperation<BaseStorageFile?>`

File: src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs [hunk 7]
```diff
@@ -112,23 +112,20 @@ public static async Task<bool> CheckDefaultZipApp(string filePath)
 
 		public static IAsyncOperation<BaseStorageFolder> FromPathAsync(string path)
```
Comment (hez2010) [inline src/Files.App/Filesystem/StorageItems/ZipStorageFolder.cs:113]: Nullable: `IAsyncOperation<BaseStorageFolder?>`

**Review by hez2010 — APPROVED** [review]
Can you also update the nullable type annotations?
Otherwise LGTM.

**Review by yaira2 — APPROVED** [review]
LGTM

**Pull request merged**